server/service/hosts.go

```diff
@@ -722,6 +722,18 @@ func (svc *Service) getHostDetails(ctx context.Context, host *fleet.Host, opts f
 		return nil, ctxerr.Wrap(ctx, err, "get batteries for host")
 	}
 
+	// Due to a known osquery issue with M1 Macs, we are ignoring stored in the db and replacing it
+	// the service layer with custom values determined by the cycle count.
```

Comment:
quick question: why at the service layer? just wondering in case we need to use `ListHostBatteries` somewhere else.

one option would be to bake this in the SQL query:

```sql
    SELECT
      host_id,
      serial_number,
      cycle_count,
      CASE
        WHEN cycle_count < 1000 THEN "Normal"
        ELSE "Replacement recommended"
      END AS health
    FROM
      host_batteries
    WHERE
      host_id = ?
```

Reply:
Thanks! I'll go ahead and merge so they can get the patch release out and make a follow up ticket to move this to the db layer so we can make broader use of `ListHostBatteries` in the future.